mitodl/mitxonline
# Convert project to Typescript

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #24 
Closes #40

#### What's this PR do?

This PR dominoed into a set of dependent changes, hence the large diff:

- Initial conversion to typescript
  - Generally a lot of cleanup to be done as `flow` was not as strict despite being inflexible a lot of the time
- Refactored HOC components to function based to make the typing easier
  - Refactored several things into hooks to make the ergonomics better, particularly the notifications system.
- Upgraded webpack to support typescript
- Upgraded code formatters and linters to support typescript
- Switch to `stylelint` instead of `sass-lint` because newest `prettier` was causing the latter to fail.

#### How should this be manually tested?
Go through the app and make sure everything still works, generally:
- Able to register a new account
- Able to login to an existing account
- Able to update email/password
- Able to update profile
- Able to enroll in a course
- Able to interact with the dashboard options per-course
